ui/app/routes/vault/cluster/auth.js

```diff
@@ -25,19 +25,30 @@ export default class AuthRoute extends ClusterRouteBase {
   }
 
   async model(params) {
+    const clusterModel = this.modelFor('vault.cluster');
     const wrapped_token = params?.wrapped_token;
+    // logs user in directly via URL query param
     if (wrapped_token) {
-      const clusterModel = this.modelFor('vault.cluster');
-      await this.unwrapToken(wrapped_token, clusterModel.id);
+      const authResponse = await this.unwrapToken(wrapped_token, clusterModel.id);
+      return { clusterModel, unwrapResponse: authResponse };
     }
-    return super.model(...arguments);
+
+    const visibleAuthMounts = await this.fetchMounts();
+    return {
+      clusterModel,
+      storedLoginData: this.auth.getAuthType(),
+      visibleAuthMounts,
+    };
   }
 
   resetController(controller) {
     controller.set('authMethod', 'token');
   }
 
-  afterModel() {
+  afterModel(model) {
```

Comment:
Moved the transition to the afterModel hook because `Error: TransitionAborted` was occasionally failing the `wrapped_token` [tests](https://github.com/hashicorp/vault/actions/runs/14816931800/job/41598645885). This way the model gets to cleanly finish its transition and then we redirect the user.

Reply:
double 🎉